## Study questions for Chapter 7 computer lab session ##

# Name:
# Date:

### Note that the tutorial that will explain most of the answers to these questions is
### here: https://markravinet.github.io/Chapter7.html

### INSTRUCTIONS ###
# The easiest way to submit your work is use this R script as a template. You should show
# your code. You should also show the answers using the hash (#) symbol to comment them
# out.
#
# When you are finished, you can click, File>Save_as and alter the extension of the script
# from .R to .txt - this will save it as a text file which you can then upload to Canvas.
# We recommend you do this last of all, because it will destroy the syntax highlighting in
# your R code.

# 1. How long are the sequences in the mydna object?

# 2. How many polymorphic positions (i.e. SNPs) are there in the mydna object? Where in the sequence do
##   they occur?

# 3. What is the standardised segregating sites for the mydna object?

# 4. How can we correct the step-by-step nucleotide diversity so that it matches the output from nuc.div for the mydna
##   object? You need to show your R code here.

## 5. How long are the sequences in the woodmouse data? What are base frequencies and what is the GC content?

## 6. Calculate the number of segregating sites (actual and standardised) and the nucleotide diversity for the woodmouse data.

## 7. How many segregating sites (i.e. actual number) are there if we subset the woodmouse dataset to 12 individuals 
##    and 500 basepairs? What is the nucleotide diversity?

## 8. The Tajima's D test we performed on the woodmouse data is not significantly different from zero. However, for our purposes
## here, imagine that it is. Recall that the data is from mitochondrial DNA (believed to be neutral) and was used in a study of the
## woodmouse's demographic history since the last ice age, where it seems likely that they survived in a refugia in Southern Europe
## and then recolonised Northern Europe following ice retreat. What is the value we calculated here and what might it represent? 
## NB. Some information from the textbook will also help you answer this question. 

## 9. How many SNP positions are there in our sparrows dataset? What proportion of these have more than two alleles?

## 10. The VCF we read in contains data from 129 individuals - how many from each species?

## 11. How many sliding windows do we generate with a window width of 100 Kb and a 25 Kb jump?

## 12. Plot the sliding window nucleotide diversity for both the spanish sparrow (using red) and 
## also the italian sparrow (using gold). Do you see the same region of reduced diversity as with the house sparrow?
## If there is a shared pattern, what might explain it?






